posts/lte-examples.md

```diff
@@ -0,0 +1,491 @@
+```lean
+example [fact (0 < p)] [fact (p ≤ 1)] 
+  (f g : pBanach.lp p) (n : ℕ) :
+  (f + g) n = f n + g n := rfl
+
+example [fact (0 < p)] [fact (p ≤ 1)] (a : ℝ) 
+  (f : pBanach.lp p) (n : ℕ) :
+  (a • f) n = a * f n := rfl
+```
+
+# The `Ext` groups
+
+The file [`examples/Ext.lean`](https://github.com/leanprover-community/lean-liquid/blob/master/src/examples/Ext.lean) was arguably the original motivation for the `examples` folder.
+After the liquid tensor experiment was completed, we joked about the fact that we could have "accidentily" defined `Ext` to always be zero!
+
+![ext-zulip](/images/lte-ext-zulip-1.png)
+![ext-zulip](/images/lte-ext-zulip-2.png)
+
+Of course, then quickly came the question of how we could be convinced that the definition we had was correct.
+
+We came up with two computations that were sufficiently convincing for us:
+
+1. We showed that our definition of `Ext` yields a universal $\delta$-functor (in the first variable).
+  Unfortunately, at the time of writing, $\delta$-functors are still not part of mathlib.
+  Their definition is in the LTE repository, and can be found [here](https://github.com/leanprover-community/lean-liquid/blob/25dc13f472934009786afaaaa2392fa2c8d73e3c/src/for_mathlib/universal_delta_functor/basic.lean#L24).
+2. We did the very first exercise one might do when first learning about Ext groups: $\operatorname{Ext}^1(\mathbb{Z}/n,\mathbb{Z}/n) \cong \mathbb{Z}/n$.
+
+## `Ext` vs. `Ext'`
+While `mathlib` does have the definition of Ext groups of two objects in an abelian category with enough projectives, for the purposes of LTE, we had to also consider Ext groups of complexes.
+We ended up making a new definition `Ext i A B` where `A` and `B` are arbitrary bounded above complexes in an abelian category with enough projectives (more precisely, we ended up working with the homotopy category of bounded above cochain complexes).
+The Ext groups for objects, denoted `Ext' i X Y` is then defined by viewing an object $X$ as the complex $X[0]$ concentrated in degree zero.
+```lean
+example (n : ℕ) (X Y : 𝓐) :
+  Ext' n (op X) Y =
+  Ext n (op ↑X) ↑Y := rfl
+```
+The presence of `↑` in this code indicates that a coercion is involved.
+In this case, it is the coercion from the abelian category `𝓐` to the bounded-above homotopy category of cochain complexes in `𝓐`, denoted `bounded_homotopy_category 𝓐` throughout the repository.
+
+## `Ext'` as a $\delta$-functor
+As aluded to in item 1 above, `Ext' i (-) Y` can be assembeled to form a (contravariant, cohomological) $\delta$-functor.
+```lean
+example (Y : 𝓐) : 𝓐ᵒᵖ ⥤δ Ab.{v} := Ext_δ_functor 𝓐 Y
+
+example (n : ℕ) (Y : 𝓐) : 𝓐ᵒᵖ ⥤ Ab.{v} := Ext_δ_functor 𝓐 Y n
+
+example (n : ℕ) (X Y : 𝓐) :
+  (Ext_δ_functor 𝓐 Y n) (op X) = Ext' n (op X) Y :=
+rfl
+```
+
+## Comparison with `Hom`
+In degree zero `Ext' 0 X Y` is isomorphic to the usual Hom functor, as expected.
+```lean
+example (X Y : 𝓐) : Ext' 0 (op X) Y ≅ AddCommGroup.of (X ⟶ Y) :=
```

Comment:
This provides the iso for objects, right? Do we have the iso as functors? (I mean the natural transformation.)

Reply:
Yes we do. But it's not as readable as this statement, I think.